src/custom-elements/src/components/popover/popover.tsx

```diff
@@ -375,8 +390,70 @@ export class Popover {
     };
     this._configurePopoverPopper();
     this._componentLoaded = true;
-    document.addEventListener('click', this._documentClickHandler);
-    document.addEventListener('keyup', this._documentKeyHandler);
+
+    if (this.drag) {
+      const popoverHeader = this._popoverElement
+        .querySelector(`.${POPOVER_CLASSES.HEADER}`) as HTMLElement;
+      let initialTransformProperty: string;
+      const dragMouseDown = (e: MouseEvent) => {
+        e.preventDefault();
+        this._closePrevented = true;
+        pos3 = e.clientX;
+        pos4 = e.clientY;
+        initialTransformProperty = window.getComputedStyle(this._popoverElement)['transform'];
+
+        document.onmouseup = closeDragElement;
+        document.onmousemove = elementDrag;
+      };
+      const dragTouch = (e: TouchEvent) => {
+        e.preventDefault();
+        this._closePrevented = true;
+        pos3 = e.touches[0].clientX;
+        pos4 = e.touches[0].clientY;
+        initialTransformProperty = window.getComputedStyle(this._popoverElement)['transform'];
+
+        popoverHeader.ontouchmove = elementDragOnTouch;
+      };
+      const setPosition = (clientX: number, clientY: number) => {
+        pos1 = pos3 - clientX;
+        pos2 = pos4 - clientY;
+        pos3 = clientX;
+        pos4 = clientY;
+        this._popoverElement.style.top = this._popoverElement.offsetTop - pos2 + "px";
+        this._popoverElement.style.left = this._popoverElement.offsetLeft - pos1 + "px";
+        this._popoverElement.style.transform = initialTransformProperty;
+      };
+      const elementDrag = (e: MouseEvent) => {
+        e.preventDefault();
+        this._destroyPopper();
+        setPosition(e.clientX, e.clientY);
+      };
+      const elementDragOnTouch = (e: TouchEvent) => {
+        e.preventDefault();
+        const touch = e.targetTouches[0];
+
+        this._destroyPopper();
+        setPosition(touch.clientX, touch.clientY);
+      };
+      const closeDragElement = () => {
+        document.onmouseup = null;
+        document.onmousemove = null;
+      };
+      let pos1 = 0,
+        pos2 = 0,
+        pos3 = 0,
+        pos4 = 0;
+
+      if (popoverHeader) {
+        popoverHeader.onmousedown = dragMouseDown;
+        popoverHeader.ontouchstart = dragTouch;
+      } else {
+        this._popoverElement.onmousedown = dragMouseDown;
```

Comment:
Why do you set up the listeners this way instead of adding Event listeners? This is not a good practice.

Reply:
This way was easier removing the event handler but I agree, addEventListener is a better approach. Updated, thanks 😉